w3af/core/data/parsers/doc/open_api/parameters.py

```diff
@@ -33,6 +35,83 @@ class OpenAPIParamResolutionException(Exception):
     pass
 
 
+class ParameterValueParsingError(Exception):
+
+    def __init__(self, message):
+        super(ParameterValueParsingError, self).__init__(message)
+
+
+class ParameterValues(object):
+
+    def __init__(self):
+        self._values = {}
+
+    def is_empty(self):
+        """
+        :return: True if no parameter values were set, false otherwise.
+        """
+        return not self._values
+
+    def get(self, path, name):
+        """
+        Gets values for a parameter of API endpoint.
+        :param path: A path to the API endpoint.
+        :param name: A name of the parameter.
+        :return: A list of values assigned to the parameter.
+        """
+        values = self._values.get(self.key(path, name), None)
+        if not values:
+            return []
+        return list(values)
+
+    def set(self, path, name, values):
+        """
+        Sets values for a parameter of API endpoint.
+        :param path: A path to the API endpoint.
+        :param name: A name of the parameter.
+        :param values: A list of values to be assigned to the parameter.
+        """
+        if not isinstance(values, list):
+            raise ValueError('values is not a list')
+        self._values[self.key(path, name)] = list(values)
+
+    @staticmethod
+    def key(path, name):
+        return '%s|%s' % (path, name)
+
+    def load_from_file(self, filename):
+        """
+        Loads parameter values from a YAML file.
+        :param filename: A path to the file to be loaded.
+        """
+        with open(filename, 'r') as content:
+            return self.load_from_string(content)
+
+    def load_from_string(self, string):
+        """
+        Loads parameter values from YAML.
+        :param string: Definition of parameter values in YAML.
+        """
+        content = yaml.load(string)
```

Comment:
User might enter an invalid YAML, that case should be handled.

Reply:
Okay, I'll add a try-except block to catch `YAMLError` and wrap the exception in to `ParameterValueParsingError`.